profile_library/lidl/manufacturer.json

```diff
@@ -3,6 +3,7 @@
   "aliases": [
     "_TZ3000_riwp3k79",
     "_TZ3210_hxtfthp5",
-    "_TZ3000_quqaeew6"
+    "_TZ3000_quqaeew6",
+    "_TZ3000_th6zqqy6"
```

Comment:
What are the aliases? Are these really manufacturer aliases or model aliases?

Reply:
These are really manufacturer aliases. Lidl uses those aliases to differentiate different models, so say their newest generation of bulbs get one of them instead of 'lidl' and the model name is then just the chip name.

See for example [here](https://github.com/Koenkk/zigbee-herdsman-converters/blob/ba97340d2416dfde77794d780fec9c44d403fa90/src/devices/lidl.ts#L211)

Reply:
> These are really manufacturer aliases. Lidl uses those aliases to differentiate different models, so say their newest generation of bulbs get one of them instead of 'lidl' and the model name is then just the chip name.
> 
> See for example [here](https://github.com/Koenkk/zigbee-herdsman-converters/blob/ba97340d2416dfde77794d780fec9c44d403fa90/src/devices/lidl.ts#L211)

Ok I see, so depending on the zigbee integration used ZHA, vs Zigbee2Mqtt etc. 
This manufacturer `_TZ3000_th6zqqy6` is exposed in HA device information instead of `lidl`?
Because in your screenshot I see it's Lidl for you, so it already seems to be correctly converted by zigbee-herdsman-converters and `_TZ3000_th6zqqy6` is never exposed in HA.
When this is the case it does not make a lot of sense to keep a separate DB of all these zigbee ID's in Powercalc library.

Reply:
Well, I don't know if ZHA exposes this as manufacturer. Just used it for a day or two for testing purposes. I know however that you can expose those as additional fields via Z2M. [We talked about this in the past.](https://github.com/bramstroker/homeassistant-powercalc/pull/368)

That's why I thought it's custom to add those to Powercalc as well.

Technically it would be better to add those not as manufacturer aliases, but do matching on the device level. Say
```
zigbee_manufacturer: "",
zigbee_device: ""
```

As explained (maybe a bit convoluted) [here](https://github.com/bramstroker/homeassistant-powercalc/issues/2663).

So in short:

Z2M does matching based on ZigBee manufacturer and ZigBee device type, and replaces both with something like “Lidl” and “HG08131B”.

But as soon as the second device is added by a manufacturer, the device type is changed to a list, to something like "[HG07834A/HG09155A](https://www.zigbee2mqtt.io/devices/HG07834A_HG09155A.html#lidl-hg07834a%252Fhg09155a)" and simple matching of the name does not work anymore.

So users are never informed, that there's a matching profile for their device.

The best solution would be, IMHO to show "hey there's maybe a profile available for your device" and then show a list of the 2-3 devices which match the identifier.

The solutions to achieve this are:

- Recommending users to set "Include device information" under MQTT in z2m, so Zigbee info is passed through as additional fields. Those are stable and could be matched.
- Separate the string list send by z2m (and ZHA) and match each of those, to search for the devices powercalc has in it's library. And if there's any match, show this to the user as option.
- Add aliases each time z2m/zha changes the string lists for the device (probably the worst one, as we need to keep updating the strings all the time).

I recommend a dual approach: Add ZigBee device type and manufacturer to the device profiles and if they are present in the device, search for them. This allows the best matching, as those are stable.

If that's not available, match the manufacturer and then separate the string lists and match each of those.

Reply:
I only have access to the device information exposed in HA device info.
`manufacturer`, `model`, model_id` fields.
You can also check these fields in `.storage/core.device_registry`

I don't have access to any fields in Z2M or something like that, as that's only internal to that specific integration and unaccesable for other integrations.
Even I was able to access that in a very hacky way it won't happen as it's prone to break, and I don't want to have code specific to other integrations in Powercalc as that would become a maintenance nightmare.

I need to think about your proposal about multiple profiles matching a certain model.
But before proposing any further solutions I wanted to indicate which information is available to Powercalc to work with. Which is only the device info available in HA.

Reply:
@bramstroker there's no need to hack anything. Z2M has an option to extend the device specifications. It's called "Include device information".

Then the Zigbee info should become available in HA (as far as I understand). Here's how those message fields look like:

![Screenshot_20241112_205605](https://github.com/user-attachments/assets/d3569ab5-88ef-4266-b940-d80eee5b53a5)

The whole idea was: A simple match for such a string is simpler than to parse a dash separated list, and less error-prone. But both would work.

Note that (as far as I understand) device discovery might change in the future in Z2M to new device discovery messages:

https://github.com/Koenkk/zigbee2mqtt/discussions/24198#discussioncomment-11170411

Reply:
> @bramstroker there's no need to hack anything. Z2M has an option to extend the device specifications. It's called "Include device information".
> 
> Then the Zigbee info should become available in HA (as far as I understand). Here's how those message fields look like:
> 
Did you actually check HA device info as I indicated? Or is this an assumption. As I'm not so sure about it this information is exposed in HA device info. I don't use Zigbee myself so can't check.

> Note that (as far as I understand) device discovery might change in the future in Z2M to new device discovery messages:
> 
> [Koenkk/zigbee2mqtt#24198 (comment)](https://github.com/Koenkk/zigbee2mqtt/discussions/24198#discussioncomment-11170411)

This seems particularly related to how MQTT devices are discovered, how device discovery messages are sent.

Reply:
I just checkt the option @RubenKelevra mentioned above. There is no change in the discovery messages.
A quick Google search revealed that this option actually is suppose to turn on including exactly the information that is already included in the mqtt messages. The checkbox has no effect at all and would actually have to be checked anyway to make powercalc work with zigbee devices. But since the manufacturer and model is always included it does not matter. Don't forget, the zigbee webpage is only a frontend! In the end the backend is what is deciding what happens. It seem strange maybe but I also had to find out searching to solve a problem that this is not always the same.

Additionally, the zigbee manufacturer @RubenKelevra shows in the screenshot will never be exposed in the mqtt messages at the moment.

Reply:
> Additionally, the zigbee manufacturer @RubenKelevra shows in the screenshot will never be exposed in the mqtt messages at the moment.

How can I screenshot those then? ;)

I've turned the option off and those device infos disappear. Switch does work fine.

![Screenshot_20241112_231554](https://github.com/user-attachments/assets/300fa081-a0fb-412c-87eb-f5b7cc404ce1)

This also has nothing to do with the GUI, as the switch is documented in the z2m documentation:

https://www.zigbee2mqtt.io/guide/configuration/mqtt.html

Reply:
My checkbox was off and I got the same information in the message as you do.

But the information you took a screenshot of is not the discovery message it is a sensor payload. This information is not used and exposed for the device. It is moste likely not even part of the attributes of a sensor.

The foremost important information for a (new) model is a screenshot of the device information of the device page and what it shows there.

Actually, the right way to address the problem is the project zigbee2mqtt. Each supported device has a configuration and this needs to be altered to not just show "Mueller light" for every different light there is even the zigee model name is different but actually a different name for each model or the string you took a screenshot of.
As far as I tracked the development of zigbee2mqtt just due to the flood of models added by the manufacturers most of the times someone just copy and pasts n existing configuration alters the identifier and done. If someone is not experienced this might lead to your problem.

Reply:
Just a short addition to the zigbee2mqtt problem:

I have several devices where I can actually see the issue I described above. I'm my case not because of the naming but more problematic, someone has copied the configuration, changed the identifiers and done. This person did not have the device and the result is a flawed control where the controls of the light behave not 100% correct but in general they do.

Reply:
> Actually, the right way to address the problem is the project zigbee2mqtt. Each supported device has a configuration and this needs to be altered to not just show "Mueller light" for every different light there is even the zigee model name is different but actually a different name for each model or the string you took a screenshot of.
> As far as I tracked the development of zigbee2mqtt just due to the flood of models added by the manufacturers most of the times someone just copy and pasts n existing configuration alters the identifier and done.

I've added the information in the z2m project today with a PR.

But this does not fix the described issue, that if more than one light got the same model/vendor string that z2m is forced to create a dash delimited string-list, which changes over time. Either we split the string list and try to match the models, or we read the ZigBee model/vendor string directly.

I thought matching the zigbee vendor/device string is a more robust solution, but if you're saying it doesn't end up in parsable form in HA, what's the purpose of this option in Z2M then? That's kinda weird.

Maybe a PR here on z2m would make sense, so the ZigBee vendor/device string is always added to devices to their specifications/diagnostic info? That would make matching more robust.

> If someone is not experienced this might lead to your problem.

I suggest that you watch your tone.

Reply:
> 
> > If someone is not experienced this might lead to your problem.
> 
> I suggest that you watch your tone.

@RubenKelevra I am sorry to have offended you by shortening my paragraph. It seems you take the issue we discuss rather personal. This sentence was not pointed at you but in general. In z2m you can create multiple configurations that lead to duplicate manufacturer and model in HA as you yourself pointed out. But you can also make them distinguishable even though they are pointing to the same configuration file in z2m. 
To elaborate my point again: in z2m often added support to new devices is done by adding an identifier to a existing configuration because all the logic is the same. A second fast option is just copying an existing configuration and just changing some details. In both cases this easily leads to the issue we discuss here.

Reply:
> > Actually, the right way to address the problem is the project zigbee2mqtt. Each supported device has a configuration and this needs to be altered to not just show "Mueller light" for every different light there is even the zigee model name is different but actually a different name for each model or the string you took a screenshot of.
> > As far as I tracked the development of zigbee2mqtt just due to the flood of models added by the manufacturers most of the times someone just copy and pasts n existing configuration alters the identifier and done.
> 
> I've added the information in the z2m project today with a PR.
I guess you created a pull request that changes model and not zigbee_model? That is great!

> But this does not fix the described issue, that if more than one light got the same model/vendor string that z2m is forced to create a dash delimited string-list, which changes over time. Either we split the string list and try to match the models, or we read the ZigBee model/vendor string directly.
I guess, if the z2m project offers a king of list option in the configuration for the model string (not the zigbee_model) where the list is concatenated in a defined way, this could be used. On the other hand if this is just a string "anyone" can define freely (one person uses "/" another "," and so on to list all the models) it would not make sense to bother.

> I thought matching the zigbee vendor/device string is a more robust solution, but if you're saying it doesn't end up in parsable form in HA, what's the purpose of this option in Z2M then? That's kinda weird.
z2m is not only used by HA as you know. It is a general tool and other projects use it differently. I am not saying that it wouln't be nice to add it to the discovery payload. For that z2m would need to implement this (I huess) and HA would need to implement the exposure of this. Complicated to coordinate and agree on, I think.

> Maybe a PR here on z2m would make sense, so the ZigBee vendor/device string is always added to devices to their specifications/diagnostic info? That would make matching more robust.
*thumbs-up*